components/action-mgt/org.wso2.carbon.identity.action.execution/src/main/java/org/wso2/carbon/identity/action/execution/ActionExecutorServiceImpl.java

```diff
@@ -85,26 +85,77 @@ public boolean isExecutionEnabled(ActionType actionType) {
     }
 
+    /**
+     * Resolve the actions that need to be executed for the given action types and execute them.
+     *
+     * @param actionType    Action Type.
+     * @param eventContext  The event context of the corresponding flow.
+     * @param tenantDomain  Tenant domain.
+     * @return Action execution status.
+     */
     public ActionExecutionStatus execute(ActionType actionType, Map<String, Object> eventContext, String tenantDomain)
             throws ActionExecutionException {
 
         try {
             List<Action> actions = getActionsByActionType(actionType, tenantDomain);
             validateActions(actions, actionType);
-            ActionExecutionRequest actionRequest = buildActionExecutionRequest(actionType, eventContext);
-            ActionExecutionResponseProcessor actionExecutionResponseProcessor = getResponseProcessor(actionType);
-
-            Action action = actions.get(0); // As of now only one action is allowed.
-            return Optional.ofNullable(action)
-                    .filter(activeAction -> activeAction.getStatus() == Action.Status.ACTIVE)
-                    .map(activeAction -> executeAction(activeAction, actionRequest, eventContext,
-                            actionExecutionResponseProcessor))
-                    .orElse(new ActionExecutionStatus(ActionExecutionStatus.Status.FAILED, eventContext));
+            // As of now only one action is allowed.
+            Action action = actions.get(0);
+            return execute(action, eventContext);
         } catch (ActionExecutionRuntimeException e) {
             // todo: add to diagnostics
             LOG.debug("Skip executing actions for action type: " + actionType.name(), e);
             return new ActionExecutionStatus(ActionExecutionStatus.Status.FAILED, eventContext);
+        }
+    }
+
+    /**
+     * Resolve the actions by given the action id list and execute them.
+     *
+     * @param actionType    Action Type.
+     * @param actionIds     Lis of action Ids of the actions that need to be executed.
+     * @param eventContext  The event context of the corresponding flow.
+     * @param tenantDomain  Tenant domain.
+     * @return Action execution status.
+     */
+    public ActionExecutionStatus execute(ActionType actionType, String[] actionIds, Map<String, Object> eventContext,
+                                         String tenantDomain) throws ActionExecutionException {
 
+        try {
+            // As of now only one action is allowed.
+            if (actionIds.length != 1) {
+                throw new ActionExecutionException("Only support list with single action Id.");
+            }
+            return execute(getActionByActionId(actionType, actionIds[0], tenantDomain), eventContext);
+        } catch (ActionExecutionRuntimeException e) {
+            // todo: add to diagnostics
+            LOG.debug("Skip executing actions for action type: " + actionType.name(), e);
```

Comment:
It's fine to log an error here as this is a real error case.
Let's follow the same pattern similar to https://github.com/wso2/carbon-identity-framework/blob/b7e7c8d2d8339f0b99e57ded2b7244160e18d365/components/action-mgt/org.wso2.carbon.identity.action.execution/src/main/java/org/wso2/carbon/identity/action/execution/ActionExecutorServiceImpl.java#L168-L170

Reply:
Improved this to be done with validateActionIdList() methods.